ppl/src/main/java/org/opensearch/sql/ppl/PPLService.java

```diff
@@ -82,17 +86,31 @@ private PhysicalPlan plan(PPLQueryRequest request) {
     // 1.Parse query and convert parse tree (CST) to abstract syntax tree (AST)
     ParseTree cst = parser.parse(request.getRequest());
     UnresolvedPlan ast = cst.accept(
-        new AstBuilder(new AstExpressionBuilder(), request.getRequest()));
-
+        new AstBuilder(new AstExpressionBuilder(), catalogService, request.getRequest()));
     LOG.info("[{}] Incoming request {}", LogUtils.getRequestId(), anonymizer.anonymizeData(ast));
 
+    StorageEngine storageEngine =
```

Comment:
the catalogService should decide what is storage engine based on connector info.

Reply:
I am doing the same...I am making getConnector(ast) call in next line.